### ISSUE TITLE
Agree on some coding style

### ISSUE THREAD

**Issue by agarny** [cellml/libcellml-archived#16]
I just came across https://github.com/codecurve/libcellml/tree/buildBotDummy01 and, sure, the code is still in its infancy, but that's the reason we probably ought to agree on some coding style. For example, looking at this [version of a file](https://github.com/codecurve/libcellml/blob/9b44a2646573c9f15e6dc9753b1ae0f396c91060/src/Version.cpp), line 6 reads:

```
using namespace std;
```

of which I am not a fan at all (it can make it difficult to read some code).

Anyway, I am not claiming that I have the perfect coding style, but it might be worth looking at the [coding style used in OpenCOR](http://opencor.ws/developer/develop/codingStyle.html). As can be seen, it's pretty much the coding style used by the Qt Creator team.

**Comment by codecurve** [comment]
This looks good, though have only had a brief look so far: http://google-styleguide.googlecode.com/svn/trunk/cppguide.html

**Comment by nickerso** [comment]
I'd suggest adapting the OpenCOR style to libCellML would probably be a good starting point.

**Comment by codecurve** [comment]
As discussed at today's meeting, this issue is a sub-task of #42

**Comment by codecurve** [comment]
I've spent some more time looking over a few C++ style guides, including the OpenCOR style again.  The OpenCOR style has Qt specific aspects, and it is not as comprehensive as the Google style guide.  I've also looked over some other style guides, but my preference is still for the Google style, with minor modifications: 
- variable names using camel case for variable names, starting on a lower-case letter (rather than all lower-case separated by underscores).  
- line length of 160 characters.
- regular function names start on lower-case letter.
- Doxygen comments
